src/protocols/secure_channel/PASESession.cpp

```diff
@@ -700,56 +789,65 @@ CHIP_ERROR PASESession::HandleMsg3(const System::PacketBufferHandle & msg)
 
     if (err != CHIP_NO_ERROR)
     {
-        SendErrorMsg(spake2pErr);
+        SendStatusReport(spake2pErr);
     }
     return err;
 }
 
-void PASESession::SendErrorMsg(Spake2pErrorType errorCode)
+void PASESession::SendStatusReport(uint16_t protocolCode)
 {
-    System::PacketBufferHandle msg;
-    uint16_t msglen        = sizeof(Spake2pErrorMsg);
-    Spake2pErrorMsg * pMsg = nullptr;
+    GeneralStatusCode generalCode =
+        (protocolCode == kProtocolCodeSuccess) ? GeneralStatusCode::kSuccess : GeneralStatusCode::kFailure;
+    uint32_t protocolId = Id.ToFullyQualifiedSpecForm();
+
+    ChipLogDetail(SecureChannel, "Sending status report");
 
-    msg = System::PacketBufferHandle::New(msglen);
-    VerifyOrReturn(!msg.IsNull(), ChipLogError(SecureChannel, "Failed to allocate error message"));
+    StatusReport statusReport(generalCode, protocolId, protocolCode);
 
-    pMsg        = reinterpret_cast<Spake2pErrorMsg *>(msg->Start());
-    pMsg->error = errorCode;
+    Encoding::LittleEndian::PacketBufferWriter bbuf(System::PacketBufferHandle::New(statusReport.Size()));
+    statusReport.WriteToBuffer(bbuf);
 
-    msg->SetDataLength(msglen);
+    System::PacketBufferHandle msg = bbuf.Finalize();
+    VerifyOrReturn(!msg.IsNull(), ChipLogError(SecureChannel, "Failed to allocate status report message"));
 
-    VerifyOrReturn(mExchangeCtxt->SendMessage(Protocols::SecureChannel::MsgType::PASE_Spake2pError, std::move(msg)) ==
-                       CHIP_NO_ERROR,
-                   ChipLogError(SecureChannel, "Failed to send error message"));
+    VerifyOrReturn(mExchangeCtxt->SendMessage(MsgType::StatusReport, std::move(msg)) == CHIP_NO_ERROR,
+                   ChipLogError(SecureChannel, "Failed to send status report message"));
```

Comment:
This code should be common across all protocol handlers in Secure channel. Not in this PR, but it would be worthwhile to factor this out perhaps as a function that takes an exchange context and the rest of the basic args and sends a basic status report. The pattern is the same everywhere.

Reply:
Added a TODO to move it to common part of the code.